src/apps/testapps/testPolygonToCellsExperimental.c

```diff
@@ -0,0 +1,488 @@
+}
+
+static void fillIndex_assertions(H3Index h) {
+    if (isTransmeridianCell(h)) {
+        // TODO: these do not work correctly
+        return;
+    }
+
+    int currentRes = H3_EXPORT(getResolution)(h);
+    // TODO: Not testing more than one depth because the assertions fail.
+    for (int nextRes = currentRes; nextRes <= currentRes + 1; nextRes++) {
+        CellBoundary bndry;
+        H3_EXPORT(cellToBoundary)(h, &bndry);
+        GeoPolygon polygon = {
+            .geoloop = {.numVerts = bndry.numVerts, .verts = bndry.verts},
+            .numHoles = 0,
+            .holes = 0};
+
+        int64_t polygonToCellsSize;
+        t_assertSuccess(H3_EXPORT(maxPolygonToCellsSize)(&polygon, nextRes, 0,
+                                                         &polygonToCellsSize));
+        H3Index *polygonToCellsOut =
+            calloc(polygonToCellsSize, sizeof(H3Index));
+        t_assertSuccess(H3_EXPORT(polygonToCellsExperimental)(
+            &polygon, nextRes, 0, polygonToCellsOut));
+
+        int64_t polygonToCellsCount =
+            countNonNullIndexes(polygonToCellsOut, polygonToCellsSize);
+
+        int64_t childrenSize;
+        t_assertSuccess(
+            H3_EXPORT(cellToChildrenSize)(h, nextRes, &childrenSize));
+        H3Index *children = calloc(childrenSize, sizeof(H3Index));
+        t_assertSuccess(H3_EXPORT(cellToChildren)(h, nextRes, children));
+
+        int64_t cellToChildrenCount =
+            countNonNullIndexes(children, childrenSize);
+
+        t_assert(polygonToCellsCount == cellToChildrenCount,
+                 "PolygonToCells count matches cellToChildren count");
+
+        for (int i = 0; i < childrenSize; i++) {
+            bool found = false;
+            if (children[i] == H3_NULL) continue;
+            for (int j = 0; j < polygonToCellsSize; j++) {
+                if (polygonToCellsOut[j] == children[i]) {
+                    found = true;
+                    break;
+                }
+            }
+            t_assert(
+                found,
+                "All indexes match between polygonToCells and cellToChildren");
+        }
+
+        free(polygonToCellsOut);
+        free(children);
+    }
+}
+
+SUITE(polygonToCells) {
+    sfGeoPolygon.geoloop = sfGeoLoop;
+    sfGeoPolygon.numHoles = 0;
+
+    holeGeoPolygon.geoloop = sfGeoLoop;
+    holeGeoPolygon.numHoles = 1;
+    holeGeoPolygon.holes = &holeGeoLoop;
+
+    emptyGeoPolygon.geoloop = emptyGeoLoop;
+    emptyGeoPolygon.numHoles = 0;
+
+    invalidGeoPolygon.geoloop = invalidGeoLoop;
+    invalidGeoPolygon.numHoles = 0;
+
+    invalid2GeoPolygon.geoloop = invalid2GeoLoop;
+    invalid2GeoPolygon.numHoles = 0;
+
+    pointGeoPolygon.geoloop = pointGeoLoop;
+    pointGeoPolygon.numHoles = 0;
+
+    lineGeoPolygon.geoloop = lineGeoLoop;
+    lineGeoPolygon.numHoles = 0;
+
+    // --------------------------------------------
+    // maxPolygonToCellsSize
+    // --------------------------------------------
```

Comment:
These tests are duplicated. My intent was to have copy/paste versions of the old tests with the new function, so that when we drop the old function it would just be removing the `-Experimental` files. But I guess you're right that there's no point duplicating the `maxPolygonToCellsSize` tests, I can remove.